indico/modules/events/timetable/controllers/display.py

```diff
@@ -73,28 +80,64 @@ def _process(self):
         return jsonify(html=html)
 
 
+@dataclass(frozen=True)
+class TimetableExportConfig:
+    show_title: bool
+    show_affiliation: bool
+    show_cover_page: bool
+    show_toc: bool
+    show_session_toc: bool
+    show_abstract: bool
+    dont_show_poster_abstract: bool
+    show_contribs: bool
+    show_length_contribs: bool
+    show_breaks: bool
+    new_page_per_session: bool
+    show_session_description: bool
+    print_date_close_to_sessions: bool
+
+
 class RHTimetableExportPDF(RHTimetableProtectionBase):
-    def _process(self):
+    @use_kwargs({'download': fields.Bool(load_default=False)}, location='query')
+    def _process(self, download):
         form = TimetablePDFExportForm(formdata=request.args, csrf_enabled=False)
+
         if form.validate_on_submit():
-            form_data = form.data_for_format
-            pdf_format = TimetablePDFFormat(form_data)
-            if not form.advanced.data:
-                pdf_format.contribsAtConfLevel = True
-                pdf_format.breaksAtConfLevel = True
-                pdf_class = SimplifiedTimeTablePlain
-                additional_params = {}
-            else:
-                pdf_class = TimeTablePlain
-                additional_params = {'firstPageNumber': form.firstPageNumber.data,
-                                     'showSpeakerAffiliation': form_data['showSpeakerAffiliation'],
-                                     'showSessionDescription': form_data['showSessionDescription']}
-            if request.args.get('download') == '1':
-                pdf = pdf_class(self.event, session.user, sortingCrit=None, ttPDFFormat=pdf_format,
-                                pagesize=form.pagesize.data, **additional_params)
-                return send_file('timetable.pdf', BytesIO(pdf.getPDFBin()), 'application/pdf')
+            days = {}
+            now = now_utc()
+            css = render_template('events/timetable/pdf/timetable.css')
+
+            for day in self.event.iter_days():
+                entries = (self.event.timetable_entries
+                           .filter(
+                               db.cast(TimetableEntry.start_dt.astimezone(self.event.tzinfo), db.Date) == day,
+                               TimetableEntry.parent_id.is_(None))
+                           .order_by(TimetableEntry.start_dt))
```

Comment:
```suggestion
                           .order_by(TimetableEntry.start_dt)
                           .all())
```

Otherwise you assign a query object which gets executed when you iterate over it - and you do so multiple times so probably it also gets executed multiple times. Generally, we do not want to pass query objects to templates since usually you expect lists etc there but not query objects.

Reply:
Outdated after one of your other comments